### PR TITLE
Create YAML deployment file on release

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

In this PR, I merged the `pkg/resources/manifest/deploy.yaml` and `pkg/resources/crd/bases/gadget.kinvolk.io_traces.yaml` with `IMAGE_TAG` replacing to create a YAML file which can be used to deploy Inspektor Gadget.
The file was generated here:
https://github.com/eiffel-fl/inspektor-gadget/suites/7526838885/artifacts/310062607

Best regards.

### CODE REVIEW

**Review by mauriciovasquezbernal — COMMENTED** [review]
One concern (as I already mentioned in https://github.com/kinvolk/inspektor-gadget/pull/755#discussion_r924487057) that I have with this approach is that the files on the repo will always point to latest, i.e., `kubectl apply -f https://raw.githubusercontent.com/kinvolk/inspektor-gadget/v0.8.0/pkg/resources/manifests/deploy.yaml` will use the latest image instead of v0.8.0.

I feel we should update those files before tagging the release, but then, is the release artifact still useful? 

@alban @blanquicet @flyth any opinions on this?

File: .github/workflows/inspektor-gadget.yml [hunk 1]
```diff
@@ -444,11 +444,31 @@ jobs:
...
+        TMP2=${TMP1#*/}
+        IMAGE_TAG=${TMP2//\//-}
+        if [ "$IMAGE_TAG" = "main" ]; then
+            IMAGE_TAG="latest"
+        fi
+        echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
+    - name: Build release YAML
+      run: |
+        # Use echo of cat to avoid printing a new line between files.
+        echo "$(cat pkg/resources/manifests/deploy.yaml) $(cat pkg/resources/crd/bases/gadget.kinvolk.io_traces.yaml)" > inspektor-gadget.yaml
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply .github/workflows/inspektor-gadget.yml:464]: What's the issue with an empty line there?

**Comment by mauriciovasquezbernal** [comment]
> I am sure we will forget to update the files before a release, so not using an automated way is error prone.

If we define a release procedure I don't think we'll forget it. As mentioned above, many other projects have to update different things before the release, so we can also make it. Anyway, let's wait for other people comments here.

**Comment by alban** [comment]
> One concern (as I already mentioned in https://github.com/kinvolk/inspektor-gadget/pull/755#discussion_r924487057) that I have with this approach is that the files on the repo will always point to latest, i.e., kubectl apply -f https://raw.githubusercontent.com/kinvolk/inspektor-gadget/v0.8.0/pkg/resources/manifests/deploy.yaml will use the latest image instead of v0.8.0.

In my opinion, that's not a problem since we don't have any documentation suggesting to use such command. If that's a concern, we cpi;d document using the release URL instead of the file from the git tag.

> If we define a release procedure I don't think we'll forget it. As mentioned above, many other projects have to update different things before the release, so we can also make it. Anyway, let's wait for other people comments here.

If we want to go this route, one example of release procedure and script creating a commit to change all the versions:
https://github.com/rkt/rkt/blob/master/scripts/bump-release
https://github.com/rkt/rkt/blob/master/Documentation/devel/release.md#release-process

File: .github/workflows/inspektor-gadget.yml [hunk 2]
```diff
@@ -444,11 +444,31 @@ jobs:
...
+        export IMAGE="${{ env.REGISTRY }}/${{ env.RELEASE_CONTAINER_REPO }}:${IMAGE_TAG}"
+        perl -pi -e 's@(image:) ".+\"@$1 "$ENV{IMAGE}"@; s@"latest"@"$ENV{IMAGE_TAG}"@;' inspektor-gadget.yaml
+    - name: Add inspektor-gadget.yaml as artifact.
+      uses: actions/upload-artifact@master
+      with:
+        name: inspektor-gadget-yaml
+        path: /home/runner/work/inspektor-gadget/inspektor-gadget/inspektor-gadget.yaml
```
Comment (alban) [inline .github/workflows/inspektor-gadget.yml:471]: I think this creates an artifact of the build but not an artifact of the release.

See the actions `csexton/release-asset-action@v2` and `actions/upload-release-asset@v1.0.1` below for the differences

**Comment by blanquicet** [comment]
> > If we define a release procedure I don't think we'll forget it. As mentioned above, many other projects have to update different things before the release, so we can also make it. Anyway, let's wait for other people comments here.
>
I would prefer to go in this direction and tag update the files before tagging the release but it is important to try to automate as much as possible. For instance, I really like the example of release-process that Alban shared, the script and especially the documentation:

> If we want to go this route, one example of release procedure and script creating a commit to change all the versions: https://github.com/rkt/rkt/blob/master/scripts/bump-release https://github.com/rkt/rkt/blob/master/Documentation/devel/release.md#release-process

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM from code inspection. Thanks!

**Pull request merged**